File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = null;
		
		int n = Integer.parseInt(br.readLine());
		int[][] tri = new int[n][n];
		int[][] arr = new int[n][n];
		
		for(int i = 0; i < n; i++) {
			st = new StringTokenizer(br.readLine());
			for (int j = 0; j < i+1; j++) {
				tri[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		
		arr[0][0] = tri[0][0];
		
		for(int i = 1; i < n; i++) {
			for (int j = 0; j < i+1; j++) {
				if(j == 0)
					arr[i][j] = arr[i-1][j] + tri[i][j];
				else
					arr[i][j] = tri[i][j] + Math.max(arr[i-1][j-1], arr[i-1][j]);
			}
		}
		
		int max = 0;
		for(int j = 0; j < n; j++)
			max = Math.max(max, arr[n-1][j]);
		
		System.out.print(max);

	}

}

File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.py
import sys

n = int(sys.stdin.readline().rstrip())
arr = []
for _ in range(n):
    num = list(map(int, sys.stdin.readline().rstrip().split()))
    arr.append(num)

for i in range(1, n):
    for j in range(len(arr[i])):
        if j == 0:
            arr[i][j] += arr[i-1][0]
        elif j == len(arr[i]) - 1:
            arr[i][j] += arr[i-1][-1]
        else:
            arr[i][j] += max(arr[i-1][j-1], arr[i-1][j])

print(max(arr[-1]))